test/Chainweb/Test/Pact/PactSingleChainTest.hs

```diff
@@ -452,6 +457,67 @@ compactionIsIdempotent rdb =
                   putStrLn ""
       assertFailure "pact state check failed"
 
+compactionTransactionIndex :: ()
+  => RocksDb
+  -> TestTree
+compactionTransactionIndex rdb =
+  compactionSetup "compactionTransactionIndex" rdb testPactServiceConfig $ \cr -> do
+    setOneShotMempool cr.mempoolRef goldenMemPool
+
+    let makeTx :: Int -> BlockHeader -> IO ChainwebTransaction
+        makeTx nth bh = buildCwCmd testVersion
+          $ set cbSigners [mkEd25519Signer' sender00 [mkGasCap, mkTransferCap "sender00" "sender01" 1.0]]
+          $ setFromHeader bh
+          $ mkCmd (sshow (nth, bh))
+          $ mkExec' "(coin.transfer \"sender00\" \"sender01\" 1.0)"
+
+    supply <- newIORef @Int 0
+    madeTx <- newIORef @Bool False
+    putStrLn "" -- get around tasty formatting
+    let run = do
+          setMempool cr.mempoolRef $ mempty {
+            mpaGetBlock = \_ _ _ _ bh -> do
+              madeTxYet <- readIORef madeTx
+              if madeTxYet
+              then do
+                pure mempty
+              else do
+                n <- atomicModifyIORef' supply $ \a -> (a + 1, a)
+                tx <- makeTx n bh
+                writeIORef madeTx True
+                pure $ V.fromList [tx]
+          }
+          void $ runBlock cr.pactQueue cr.blockDb second
+          writeIORef madeTx False
+          readIORef supply >>= \s -> when (s `mod` 100 == 0) $ putStrLn $ "ran block " ++ show s
+
+    -- Run enough blocks to where TransactionIndex keeps its full history
+    replicateM_ (int C.transactionIndexKeepDepth) run
```

Comment:
Why not make this a configurable parameter to compaction, so that this test is faster?

Reply:
I didn't think about that. That makes sense.